src/test/denialofservice_tests.cpp

```diff
@@ -314,10 +316,27 @@ static CTransactionRef RandomOrphan()
     return it->second.tx;
 }
 
+static void MakeNewKeyWithFastRandomContext(CKey& key)
+{
+    std::vector<unsigned char> keydata;
+    do {
+        keydata = g_insecure_rand_ctx.randbytes(32);
+        key.Set(keydata.data(), keydata.data() + keydata.size(), /*fCompressedIn*/ true);
+    } while (!key.IsValid());
```

Comment:
nit: This can simply say
```suggestion
    assert(key.IsValid());
```

Reply:
Wouldn't it fail the test if the generated key is invalid instead of retrying?

Reply:
Yes, but that should never happen, because the test is deterministic